datafusion/sql/src/unparser/plan.rs

```diff
@@ -582,9 +558,89 @@ impl Unparser<'_> {
         }
     }
 
+    /// Convert the components of a USING clause to the USING AST
+    fn join_using_to_sql(
+        &self,
+        join_conditions: &[(Expr, Expr)],
+    ) -> Result<ast::JoinConstraint> {
+        let mut idents = Vec::with_capacity(join_conditions.len());
+        for (left, right) in join_conditions {
+            match (left, right) {
+                (
+                    Expr::Column(Column {
+                        relation: _,
+                        name: left_name,
+                    }),
+                    Expr::Column(Column {
+                        relation: _,
+                        name: right_name,
+                    }),
+                ) => {
+                    if left_name != right_name {
+                        // USING is only valid when the column names are the
+                        // same, so they should never be different
+                        return not_impl_err!(
+                            "Unsupported USING with different column names"
+                        );
```

Comment:
IMO, `not_impl_err` is for some todo items. If this case never happens, I prefer to use `internal_err` here.
```suggestion
                        return internal_err!(
                            "Unsupported USING with different column names"
                        );
```

Reply:
That can happen with `ON`, but it can't happen with `USING`, right? This function is only for `USING`.

Reply:
So I thought about this some more - these two cases should be impossible for a `LogicalPlan` made from SQL (you can't do `JOIN … USING (t1.a = t1.b)`, but could be created by a user directly creating a `LogicalPlan::Join(Join { … })`.

Given that, fallback makes sense, and I'll add a comment to explain.